tests/spec/toc.py

```diff
@@ -0,0 +1,33 @@
+# Table of contents for documentation generated from specs
+
+contents = {
```

Comment:
It looks as though this exists only to impose a non-alphabetical ordering, is that right? It seems a shame to have all this duplication just for that. Could we, instead, put leading digits on the names to get the order as we want it? That reduces duplication and will (I think) make it easier for people to add things.

Reply:
I considered that, but thought that it would be more work to manage the chapter/section numbers manually.  Feel free to push back if you disagree.

Reply:
I don't think it's obvious what the right answer is. Let's leave it as is and change it if we get annoyed.